core/tests/puppeteer-acceptance-tests/blog-post-draft-blogs.js

```diff
@@ -0,0 +1,33 @@
+const acceptanceTests = require("./utility-functions/puppeteer_utils.js");
+const testConstants = require("./utility-functions/testConstants.js");
+
+
+const blogEditOptions = "e2e-test-blog-post-edit-box";
+
+async function blogDrafts_journey() {
+  const obj = await new acceptanceTests();
```

Comment:
Conceptually, this naming is quite weird. You're saying that `obj` is an "acceptance test", and that you can init(), clickOn(), etc. in it. This doesn't really make sense from a semantic/English perspective.

Part of the conventions that we need to come up with is how to name things, so "small" things like this are actually important to think through. What is "obj" exactly here? I actually thought the idea would be to do `page.clickOn()` -- you'd have a BasePage class with methods like that, and then you'd have BlogPage, etc. which subclass from BasePage and have additional methods that are tied to specific pages.

Or is the idea that `obj` represents the `actor` or `blogPostAdmin`? That could make sense too, you could do something like `blogPostAdmin.goto(BLOG_PAGE)` and `blogPostAdmin.clickOn('button', 'OK')`.

In summary, it would help to explain a bit more about what conceptual model you're going with here. As a strawman proposal, you might have, for example:

- Page classes like BlogPage, one for each page, with methods such as `clickPublishButton(), etc.
- User classes like BlogPostAdmin, with methods such as `blogPostAdmin.createNewPost()`, `blogPostAdmin.publishPost()`, etc. -- these often start by calling methods like BlogPage.visit() and then they do stuff like BlogPage.clickPublishButton() etc. Every method name here should be understandable by a non-technical person.
- The top-level test (this file) would initialize a BlogPostAdmin class and call methods on it.

That's just an idea, but maybe take a look and let me know if you see issues with this approach? We need to find a way to express this domain so that the concepts are clear to everyone.